internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -386,14 +386,16 @@ func copyDir(l *logger.Logger, from, to string, ignoreErrs bool) error {
 		}
 	}
 
+	// Try to detect if we are running with SSDs. If we are increase the copy concurrency,
+	// otherwise fall back to the default.
+	copyConcurrency := 1
```

Comment:
I changed this a bit, if I merge https://github.com/elastic/elastic-agent/pull/3653 before you merge this you'll have a conflict to resolve.